stages/org.osbuild.greenboot

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/python3
+"""
+Configure greenboot
+
+Update configuration of greenboot in /etc/greenboot/greenbot.conf.
+"""
+
+import fileinput
+import sys
+
+import osbuild.api
+
+
+SCHEMA = """
+"additionalProperties": false,
+"required": ["config"],
+"properties": {
+  "config": {
+    "additionalProperties": false,
+    "description": "greenboot config options",
+    "type": "object",
+    "properties": {
+      "monitor_services": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+      }
+    }
+  }
+}
+"""
+
+
+def main(tree, options):
+    greenboot_conf = options.get("config", {})
+    changes = greenboot_conf.copy()
+    config_file = f"{tree}/etc/greenboot/greenboot.conf"
+    key_to_conf = {
+        "GREENBOOT_MONITOR_SERVICES": "monitor_services"
+    }
+
+    # For each of the configured options, find the first non-commented out instance
+    # of the option and replace it (if necessary). If it does not already exist, append
+    # the option to the end of the file.
+    # Keys are case case sensitive.
+    with fileinput.input(files=(config_file), inplace=True) as f:
+        for line in f:
+            if line.startswith("#"):
+                sys.stdout.write(line)
+                continue
+            line_list = line.split('=')
+            if len(line_list) != 2:
+                sys.stdout.write(line)
+                continue
+            key, current = line_list
+            value = changes.pop(key_to_conf[key], None)
+            if value is None:
+                continue
+            if key == "GREENBOOT_MONITOR_SERVICES":
+                svcs = current.strip('\"').split(" ")
+                new_svcs = [ns for ns in value if ns not in svcs]
```

Comment:
I'm unclear on the filtering here.  We're filtering out old values from the list, but then we only write the new ones.  So if there's an existing array `"a b"` and we run the stage with `["a", "b", "c"]`, the result would be `"c"`.

Is the goal to append the values from the stage options to the existing config or overwrite them?

Reply:
right, I might have just missed an extend lol